pkg/oras/copy.go

```diff
@@ -0,0 +1,178 @@
+package oras
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/containerd/containerd/content"
+	"github.com/containerd/containerd/errdefs"
+	"github.com/containerd/containerd/images"
+	"github.com/containerd/containerd/log"
+	"github.com/containerd/containerd/remotes"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras-go/pkg/target"
+)
+
+// Copy copy a ref from one target.Target to a ref in another target.Target. If toRef is blank, reuses fromRef
+// Returns the root
+// Descriptor of the copied item. Can use the root to retrieve child elements from target.Target.
```

Comment:
> Can use the root to retrieve child elements from target.Target

Is there an example or test that displays how to do this? For example, obtain a list of blob descriptors (layers) referenced by a root manifest

Reply:
I had one in them, removed it for simplicity. 

We might want to add a real example. But for now, there is one straight out of containerd, see [images.Children()](https://pkg.go.dev/github.com/containerd/containerd/images#Children)

Needless to say, that takes a [content.Provider](https://pkg.go.dev/github.com/containerd/containerd@v1.5.2/content#Provider) rather than a [remotes.Fetcher](https://pkg.go.dev/github.com/containerd/containerd/remotes#Fetcher), but we actually provider [ProviderWrapper](https://github.com/oras-project/oras-go/pull/8/files#diff-7a46b4556ee5cea8c1f066d47ac99a2a0d6592107f4e27f81fb151654693c052)

Eventually, we might just want to replace `images.ChildrenHandler` with our own; one more reduction in containerd dependency.